src/com/phonebookservice/controller/ContactController.java

```diff
@@ -0,0 +1,148 @@
+package com.phonebookservice.controller;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.phonebookservice.dispatcher.EnumQueryParams;
+import com.phonebookservice.exception.BadRequestException;
+import com.phonebookservice.exception.NotFoundException;
+import com.phonebookservice.model.Contact;
+import com.phonebookservice.server.IDataAccessAdapter;
+import com.phonebookservice.util.ErrorCode;
+import com.phonebookservice.util.ErrorMessages;
+import com.phonebookservice.util.QueryParam;
+import com.phonebookservice.util.StringUtility;
+
+/**
+ * Contact controller.
+ *
+ * @author Marwa Saleh
+ */
+public final class ContactController extends AbstractController<Contact> {
+    private static ContactController singleton;
+
+    /**
+     * Initialization of contact controller.
+     *
+     * @param database the database.
+     */
+    private ContactController(final IDataAccessAdapter database) {
+        super(database);
+
+        if (database == null) {
+            throw new IllegalArgumentException(
+                    ErrorMessages.ERROR_DATABASE_NOT_FOUND);
+        }
+    }
+
+    /**
+     * Get instance of contact controller.
+     *
+     * @param database the database.
+     * @return contact controller.
+     */
+    public static ContactController getInstance(
+            final IDataAccessAdapter database) {
+        if (singleton == null) {
+            singleton = new ContactController(database);
+        }
+
+        return singleton;
+    }
+
+    /**
+     * get contact.
+     *
+     * @param contactId the contact id.
+     *
+     * @return contact the contact.
+     */
+    @Override
+    public Contact get(final Long contactId) {
+        ContactController.checkContactId(contactId);
+        final Contact contact = getDatabase().get(contactId);
```

Comment:
super.getDatabase()

Don't call any method without :- this or super or className or objectName